src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,151 @@
+        assertEquals(VALID_PASSWORD, registeredUser.getPassword());
+        assertEquals(VALID_AGE, registeredUser.getAge());
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        assertThrows(InvalidInputDataException.class, () -> registrationService.register(null));
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        User user = new User();
+        user.setLogin(null);
+        user.setPassword(VALID_PASSWORD);
+        user.setAge(VALID_AGE);
+        assertThrows(InvalidInputDataException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_nullPassword_notOk() {
+        User user = new User();
+        user.setLogin(VALID_LOGIN);
+        user.setPassword(null);
+        user.setAge(VALID_AGE);
+        assertThrows(InvalidInputDataException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_nullAge_notOk() {
+        User user = new User();
+        user.setLogin(VALID_LOGIN);
+        user.setPassword(VALID_PASSWORD);
+        user.setAge(null);
+        assertThrows(InvalidInputDataException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_shortLogin_notOk() {
+        User user = new User();
+        user.setLogin(SHORT_LOGIN);
+        user.setPassword(VALID_PASSWORD);
+        user.setAge(VALID_AGE);
+        assertThrows(InvalidInputDataException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_shortPassword_notOk() {
+        User user = new User();
+        user.setLogin(VALID_LOGIN);
+        user.setPassword(SHORT_PASSWORD);
+        user.setAge(VALID_AGE);
+        assertThrows(InvalidInputDataException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_negativeAge_notOk() {
+        User user = new User();
+        user.setLogin(VALID_LOGIN);
+        user.setPassword(VALID_PASSWORD);
+        user.setAge(NEGATIVE_AGE);
+        assertThrows(InvalidInputDataException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_ageLessThan18_notOk() {
+        User user = new User();
+        user.setLogin(VALID_LOGIN);
+        user.setPassword(VALID_PASSWORD);
+        user.setAge(AGE_LESS_THAN_18);
+        assertThrows(InvalidInputDataException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_existingUser_notOk() {
+        User existingUser = new User();
+        existingUser.setLogin(EXISTING_LOGIN);
+        existingUser.setPassword(VALID_PASSWORD);
+        existingUser.setAge(VALID_AGE);
+        Storage.people.add(existingUser);
+
+        User newUser = new User();
+        newUser.setLogin(EXISTING_LOGIN);
+        newUser.setPassword(VALID_PASSWORD);
+        newUser.setAge(VALID_AGE);
+
+        assertThrows(UserAlreadyExistsException.class, () -> registrationService.register(newUser));
+    }
```

Comment:
In the 'register_existingUser_notOk' test, consider verifying that the user count in the storage does not increase after an attempt to register an existing user.